apps/designer/app/designer/features/sidebar-left/panels/asset-manager/asset-thumbnail.tsx

```diff
@@ -67,15 +67,15 @@ export const AssetThumbnail = (asset: Asset | UploadingAsset) => {
         alignItems: "center",
         padding: "0 $2",
         position: "relative",
-
-        "&:hover button": {
-          opacity: 1,
+        "--info-trigger-visibility": "hidden",
+        "&:hover": {
```

Comment:
when button is being hovered, we define the visibility variable that will be used inside the trigger, it acts as a public API

Reply:
one problem that is still here is that this is completely untyped by ts, so maybe we should find a way to have the child component fully describe what we are passing, either by exporting the variable with valid values or a function that we can spread

Reply:
I saw this, not sure sure how this works, but can we do the same here? https://github.com/webstudio-is/webstudio-designer/blob/4164defb0818b5da9cc6ecca12abd885b78210ba/packages/design-system/src/components/control-group.tsx#L13

Reply:
This approach has still the problem with breaching Button's encapsulation, because you would define any styles on the Button, without Button explicitly allowing or being aware of the overrides, right?

Reply:
also you could go and write styles that reach into childs internal html structure if you wanted: `[`& ${Child} span`]` and by doing so create a an implicit dependency Child knows nothing about

Reply:
I general yes, but for `visibility` it seem fine to me as it just hides entire element without making any assumptions about its implementation

Reply:
But the solution with CSS variables is good too

Reply:
Yes, I am taking a stance on this case to have a single consistent solution we use everywhere. In some cases it won't be a huge deal in others it will. How do you consistently make sure at scale that everyone knows what to do?

It's hard, so if there is a solution that works well for every case, we should use it - that's how I think about it.

Reply:
Will push the same solution, just nicer written in a bit, stay tuned

Reply:
ok now its ready